File: generated_code/python/solution_4271_1.py
from typing import List

class Solution:
    def findDegrees(self, matrix: List[List[int]]) -> List[int]:
        """
        Calculates the degree of each vertex in an undirected graph 
        represented by a adjacency matrix.
        
        Time Complexity: O(n^2) where n is the number of vertices, 
                         as we iterate through every cell in the matrix.
        Space Complexity: O(1) auxiliary space (excluding the output array).
        """
        n = len(matrix)
        ans = [0] * n
        
        for i in range(n):
            # The degree of vertex i is the sum of 1s in its corresponding row.
            # In an undirected graph with no self-loops, this counts all incident edges.
            # If self-loops are possible, matrix[i][i] == 1 adds 1 to the degree.
            ans[i] = sum(matrix[i])
            
        return ans

File: generated_code/cpp/solution_4271_1.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> findDegrees(vector<vector<int>>& matrix) {
        int n = matrix.size();
        vector<int> ans(n, 0);
        
        // The degree of a vertex i in an adjacency matrix is the sum of elements in its row.
        // For an undirected graph, self-loops (matrix[i][i] == 1) contribute to the count 
        // depending on definition, but typically matrix[i][j] == 1 means an edge.
        // We iterate through the matrix and count the number of 1s in each vertex's row.
        for (int i = 0; i < n; ++i) {
            int degree = 0;
            for (int j = 0; j < n; ++j) {
                if (matrix[i][j] == 1) {
                    degree++;
                }
            }
            ans[i] = degree;
        }
        
        return ans;
    }
};